point_pattern.py

```diff
@@ -121,6 +150,20 @@ def average_nearest_neighbor_distance(points):
     """
     mean_d = 0
 
+    shortDistanceList = []
+
+    for firstPoint in points:
+        pointInList = 500
+        for secondPoint in points:
+            if firstPoint is not secondPoint:
```

Comment:
What if the points are coincident?  Check out `enumerate`.  It will let you check positional index, so you can check if you are at the same place in both lists.